elasticsearch/shard_migration.py

```diff
@@ -0,0 +1,185 @@
+logging.basicConfig(level=logging.INFO, format=log_format)
+LOG = logging.getLogger(__name__)
+
+HELP_DESCRIPTION = """
+This is a utility script for migrating shards
+in an ElasticSearch cluster from a source node to one of the target nodes.
+"""
+
+HELP_EXAMPLE = """Example: ./indices_migration.py
+ --source-node "store-01.he-eu-hel1.logs.hq"
+ --source-node-ip "10.14.0.64"
+ --target-nodes "store-05.he-eu-hel1.logs.hq"
+ --target-nodes "store-04.he-eu-hel1.logs.hq"
+"""
+
+
+def parse_args():
+    parser = ArgumentParser(
+        prog="Elasticsearch shard migration tool",
+        description=HELP_DESCRIPTION,
+        epilog=HELP_EXAMPLE,
+    )
+    parser.add_argument(
+        "-H",
+        "--host",
+        dest="es_host",
+        default="http://localhost:9200",
+        help="ElasticSearch host.",
+    )
+    parser.add_argument(
+        "-S",
+        "--source-node",
+        required=True,
+        help="Source node name to move shards from.",
+    )
+    parser.add_argument(
+        "-I",
+        "--source-node-ip",
+        help="IP address of the source node to exclude.",
+    )
+    parser.add_argument(
+        "-T",
+        "--target-nodes",
+        required=True,
+        action="append",
+        help="Comma-separated list of target node names to move shards to.",
+    )
+
+    return parser.parse_args()
+
+
+class Elasticsearch:
+    def __init__(self, host):
+        """
+        Initialize Elasticsearch client with the specified host.
+        :param host: Elasticsearch host URL.
+        """
+        self.host = host
+
+    def move_shard(self, index_data, source_node, target_nodes):
+        """
+        Move shard from the source node to one of the target nodes for a specified index.
+        :param index: Index name for which shards should be moved.
+        :param source_node: Source node from which shards should be moved.
+        :param target_node: Target nodes to which shards should be moved.
+        """
+        index = index_data["index"]
+        shard = index_data["shard"]
+
+        for target_node in target_nodes:
+            reroute_body = {
+                "commands": [
+                    {
+                        "move": {
+                            "index": index,
+                            "shard": shard,
+                            "from_node": source_node,
+                            "to_node": target_node,
+                        }
+                    }
+                ]
+            }
+
+            try:
+                LOG.info(
+                    f"Trying to move shard {shard} of index '{index}' from {source_node} to {target_node}."
+                )
+                headers = {"Content-Type": "application/json"}
+                response = requests.post(
+                    f"{self.host}/_cluster/reroute",
+                    headers=headers,
+                    data=json.dumps(reroute_body),
+                )
+                response.raise_for_status()
```

Comment:
You can generalize this with a generic `_request()` method, here's an example:
https://github.com/status-im/infra-utils/blob/454ad109c8782e48362cdd9762b72f7ea91b0b5e/consul/dnsdisc.py#L108-L127